### PR TITLE
feat: remove deprecated @Tailor and @TailorComponent

### PR DESCRIPTION
**Pull request by Falynsky**
- removed files and code related to deprecated `@Tailor` and `@TailorComponent`
- adjusted code and examples to using only `@TailiorMixin` and `@TailorMixinComponent`
- replace lints with linteo

### CODE REVIEW

File: packages/theme_tailor/example/lib/diagnosticable_lib.dart [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // ignore_for_file: directives_ordering
-
+///
```
Comment (Rongix) [inline packages/theme_tailor/example/lib/diagnosticable_lib.dart:2]:
```suggestion

```

File: packages/theme_tailor/example/test/diagnosticable_test.dart [hunk 2]
```diff
@@ -21,20 +20,19 @@ void main() {
     expect(theme, isA<DiagnosticableTree>());
 
     expect(properties.properties, [
-      isA<DiagnosticsProperty>()
+      isA<DiagnosticsProperty<dynamic>>()
```
Comment (Rongix) [inline packages/theme_tailor/example/test/diagnosticable_test.dart:23]:
```suggestion
      isA<DiagnosticsProperty>()
```

**Review by Rongix — APPROVED** [review]

File: packages/theme_tailor/example/lib/diagnosticable_lib.dart (hunk 1 above)
Comment (Falynsky) [reply packages/theme_tailor/example/lib/diagnosticable_lib.dart:2]: wymóg linteo, przegadane na slack

File: packages/theme_tailor/example/test/diagnosticable_test.dart (hunk 2 above)
Comment (Falynsky) [reply packages/theme_tailor/example/test/diagnosticable_test.dart:23]: wymóg linteo, przegadane na slack

**Review by AndriiChemer — APPROVED** [review]

**Pull request merged**